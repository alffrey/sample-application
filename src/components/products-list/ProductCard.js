import React from 'react';
import ResponsiveImage from 'core_components/responsive-image';
import Badge from 'core_components/badge';

const ProductCard = ({
    productName = '',
    price = 0,
    productImage = '',
    isSale = false,
    isExclusive = false,
    labels = {}
}) => {
    return (
        <div className="product-card">
            <h3>{productName}</h3>

            <ResponsiveImage url={productImage} />
            {isSale && <Badge type="sale" label={labels.labelSale} />}
            {isExclusive && <Badge type="exclusive" label={labels.labelExclusive} />}
            <div className="product-card__price">{price}</div>
        </div>
    );
};

export default ProductCard;
