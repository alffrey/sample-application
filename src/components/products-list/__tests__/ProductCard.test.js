import React from 'react';
import { shallow } from 'enzyme';
import ProductCard from '../ProductCard';

/**
 * Basic test case
 */
describe('components/ProductCard', () => {
    it('snapshot test', () => {
        const sampleProps = {};
        const productCardWrap = shallow(<ProductCard {...sampleProps} />);
        expect(productCardWrap).toMatchSnapshot();
    });
});
