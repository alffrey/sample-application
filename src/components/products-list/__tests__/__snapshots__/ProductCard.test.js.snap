// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ProductCard snapshot test 1`] = `ShallowWrapper {}`;
